From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 25 Mar 2009 13:35:51 +0100
Subject: [docs] document netdev_budget
Message-id: 20090325133551.5b964325@dhcp-lab-109.englab.brq.redhat.com
O-Subject: [RHEL5.4 PATCH] BZ463249: document netdev_budget
Bugzilla: 463249
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Peter Martuccelli <peterm@redhat.com>
RH-Acked-by: David Miller <davem@redhat.com>

BZ#463249
=========
https://bugzilla.redhat.com/show_bug.cgi?id=463249

Description:
============
The NAPI poll parameter netdev_budget is not documented in
kernel-docs. Since it may have a substantial effect on at least some
network loads, it should be.

Upstream Status:
===============
Merged in commit:
e9c6a586f5335938b11be0a2554af58a2860e4ef

diff --git a/Documentation/filesystems/proc.txt b/Documentation/filesystems/proc.txt
index 558d6bb..b61eccb 100644
--- a/Documentation/filesystems/proc.txt
+++ b/Documentation/filesystems/proc.txt
@@ -1363,6 +1363,13 @@ The interface  to  the  networking  parts  of  the  kernel  is  located  in
 /proc/sys/net. Table  2-3  shows all possible subdirectories. You may see only
 some of them, depending on your kernel's configuration.
 
+netdev_budget
+-------------
+
+Maximum number of packets taken from all interfaces in one polling cycle (NAPI
+poll). In one polling cycle interfaces which are registered to polling are
+probed in a round-robin manner. The limit of packets in one such probe can be
+set per-device via sysfs class/net/<device>/weight .
 
 Table 2-3: Subdirectories in /proc/sys/net 
 ..............................................................................
